src/hooks/useOrders.ts

```diff
@@ -71,6 +73,92 @@ function areThereOrders(networkId: number | undefined, userAddress: string | und
   return state[key]?.length > 0
 }
 
+export function useOrders2(): Result {
+  const { userAddress, networkId, blockNumber } = useWalletConnection()
+  const [orders, setOrders] = useSafeState<AuctionElement[]>([])
+  console.log('UO::orders', orders)
+  //  consider first state to be loading
+  const [isLoading, setIsLoading] = useSafeState<boolean>(true)
+  console.log('UO::isLoading', isLoading)
+  // start fetching from 0
+  const [offset, setOffset] = useSafeState<number>(0)
+  console.log('UO::offset', offset)
+
+  useEffect(() => {
+    let cancelled = false
+
+    const fetchOrders = async (offset: number): Promise<void> => {
+      // isLoading is the important one
+      // controls ongoing fetching chain
+      if (!userAddress || !networkId || !isLoading) return
+
+      // contract call
+      console.log('getOrdersPaginated, offste = ', offset)
+      try {
+        const { orders, nextIndex } = await exchangeApi.getOrdersPaginated({ userAddress, networkId, offset })
+        console.log('fetched orders', orders, 'nextIndex', nextIndex)
+
+        // check cancelled bool from parent scope
+        if (cancelled) return
+
+        // Apply filters (remove deleted orders)
+        const filteredOrders = filterDeletedOrders(orders)
+
+        // ensures we don't have multiple reruns for each update
+        // i.e. offset change -> render
+        //      isLoading change -> another render
+        unstable_batchedUpdates(() => {
+          // incremental update: append
+          setOrders(oldOrders => oldOrders.concat(filteredOrders))
+
+          if (!nextIndex) {
+            // no more orders left
+            // done fetching for now
+            setIsLoading(false)
+          }
+          // move offset forward, fetch new batch
+          setOffset(offset + orders.length)
+        })
+      } catch (error) {
+        console.log('Failed to fetch orders', e)
+        // TODO: inform user
+        setIsLoading(false)
+      }
+    }
+
+    fetchOrders(offset)
+
+    // if offset changed mid-fetch
+    // which happens when forceOrdersRefresh si called
+    // ignore current fetch results
+    return (): void => {
+      cancelled = true
```

Comment:
In this case `useRef` would be better (some react warning/linter told me so)?

Reply:
blame linter but also don't trust it too much

Reply:
Not saying it told me now. I saw this is the past as a best practice for having 'instance variables' like behaviour on hooks/FC is to use `useRef` rather than plain variables for reasons.
Not really needed?

Reply:
we need a variable in function scope as close as possible
so inside `useEffect` here
`useRef` would be outer scope (relatively) and would apply to all `fetchOrders` calls
whereas we need only one cancelled per `fetchOrders` call

Reply:
I don't get that linter error for some reason